### PR TITLE
Week 20 - Project Auth, Isabella Markskog & Tobias Hagberg

### PR DESCRIPTION
**Pull request by IsabellaM5**
**Link to our deployed project**
Here's the link to our live fullstack application: https://clever-wright-d2914b.netlify.app
Here's the link to our live API: https://project-auth-isma-toha.herokuapp.com/ 

**Our reflections on how this week's project turned out**
Overall we are happy with this weeks project, it went well and it was fun to do our first fullstack application from scratch 🥳 The project acted as a great reminder of different things we've been through during the boot camp, such as React Router, Redux Store, Styled Components etc. Also really good to learn about authorization and how to create logins etc. for future projects.

For the secret content on the authenticated endpoint we had initially planned to set up a fetch using axios to an API with an endpoint that randomly sends pictures. We ran into some CORS issues and didn't manage to solve them within the time frame of the project. Instead we imported json data to our deployed database and set up so that the fetch response from our API was randomized, thereby displaying different images on the frontend.

**Things we'd like to have clarified or explained in more detail**
Nothing that we can think of within the scope of this week's project. CORS is an interesting topic that we'll need to revisit at a later time hehe 😅

### CODE REVIEW

**Review by Yennie-Leek — APPROVED** [review]
The project looks really nice!💫👍
And I sure want to go traveling now😀
The code is clean and understandable. 
I noticed that when I create a new user I only have to enter a name to enter the "seceret" content and no password..  Maybe have a look at that🙂
Other than that, great work you guys!!🥳

File: README.md [hunk 1]
```diff
@@ -1,13 +1,41 @@
...
+- Set up of reducers and slices
+- Stored access token in localStorage, which was saved to Redux store
+- Made use of browserrouter, provider, switch, routes
+- Displayed an error message when the validation of user input when creating a new user fails 
+
 
 ## View it live
 
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+* Here's the link to our live fullstack application: https://clever-wright-d2914b.netlify.app
```
Comment (KarolCislowski) [inline README.md:26]: To fix it you need to create `_redirects` file in the `public` directory of your react project and paste it inside:
``` 
/* /index.html 200
```
image

File: backend/server.js [hunk 2]
```diff
@@ -1,30 +1,117 @@
...
+  password: {
+    type: String,
+    required: true,
+    minlength: 6,
+  }, 
+  accessToken: {
+    type: String,
+    default: () => crypto.randomBytes(128).toString('hex')
+  }
+})
```
Comment (KarolCislowski) [inline backend/server.js:34]: Nice model 👍🏻

File: frontend/src/components/SignInForm.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  font-size: 20px;
+  outline: none;
+  color: black;
+  border-radius: 5px;
+  cursor: pointer;
+  transition: all 0.2s ease-out;
+  margin: 10px 5px 0px 5px;
+
+  &:hover, &:focus {
+    background: rgb(229, 152, 155, 0.5);
```
Comment (KarolCislowski) [inline frontend/src/components/SignInForm.js:54]: rgba() when you use alpha chanal

File: frontend/src/components/SignInForm.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    border: solid 1px rgb(229, 152, 155);
```
Comment (KarolCislowski) [inline frontend/src/components/SignInForm.js:55]: everywhere else you use hex notation...

File: frontend/src/pages/MainPage.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        Authorization: accessToken
+      }
+    }
+
+    fetch(API_URL('travelinspo'), options)
+      .then(res => res.json())
+      .then(data => {
+        dispatch(travelInspo.actions.setTravelInspo(data[0].urls.small))
+      })
+  }, [accessToken, dispatch])
```
Comment (KarolCislowski) [inline frontend/src/pages/MainPage.js:94]: why 2 effects if you can have one? 

```jsx
useEffect(() => {
    if (!accessToken) {
      history.push('/login')
    } else {
      const options = { ...}
      ...
    }
  }, [accessToken, history, dispatch])
```

File: frontend/src/pages/MainPage.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+  
+  const handleLogOut = () => {
+    batch(() => {
+      dispatch(user.actions.setSignOut())
+      dispatch(travelInspo.actions.setSignOut())
+    })
```
Comment (KarolCislowski) [inline frontend/src/pages/MainPage.js:100]: 👍🏻

File: frontend/src/reducers/user.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+const user = createSlice ({
+  name: 'user',
+  initialState: {
+    username: null,
+    accessToken: localStorage.getItem('accessToken') || null,
```
Comment (KarolCislowski) [inline frontend/src/reducers/user.js:7]: 😻

File: frontend/src/reducers/user.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    },
+    setSignOut: (store, action) => {
+      store.username = null
+      store.accessToken = null
+      store.errors = null
+      localStorage.removeItem('accessToken')
+    }
+  }
+})
+
```
Comment (KarolCislowski) [inline frontend/src/reducers/user.js:29]: great reducer with niece approach to saving data in the localStorage 🥇

**Review by KarolCislowski — APPROVED** [review]
Great code with nice styling. I'll approve it now but please add this `_redirects` to fix this annoying Netflix 💩

**Pull request closed without merging**